Analysis/Core/src/StepTHn.cxx

```diff
@@ -258,6 +258,81 @@ Long64_t StepTHn<TemplateArray, TemplateType>::Merge(TCollection* list)
   return count + 1;
 }
 
+template <class TemplateArray, typename TemplateType>
+void StepTHn<TemplateArray, TemplateType>::Fill(Int_t istep, Double_t weight, ...)
```

Comment:
would it be possible to have the weight as part of the variable arguments (as the last one)? Then it would be easier to incorporate this into the registry.

Reply:
You cannot find out the number of arguments. So in this case it would only work if people **always** give a weight...

Reply:
Is there a better C++17/11 interface for the variable arguments? This is still C style I believe...

Reply:
you can use variadic templates / parameter packs, which would look something like this:
template<typename ...Args>
void f(Args... args) {}

Reply:
there you can infer at compile time the size via sizeof...(Args)
You can have a look at HistFiller::fillHistAny() function in the HistogramRegistry